sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -79,22 +80,33 @@ public HttpHeader get(String name) {
     }
 
     /**
-     * Get the header value for the provided header name. Null will be returned if the header
-     * name isn't found.
+     * Removes the {@link HttpHeader header} with the provided header name. {@code Null} is returned if the header
+     * isn't found.
      *
-     * @param name the name of the header to look for
-     * @return The String value of the header, or null if the header isn't found
+     * @param name the name of the header to remove.
+     * @return the header if removed, null otherwise.
+     */
+    public HttpHeader remove(String name) {
+        return headers.remove(formatKey(name));
```

Comment:
Change `headers` map to use `ConcurrentHashMap`.

Reply:
Done